docs/Documentation/Environment/Building_Packages/modules.md

```diff
@@ -5,66 +5,84 @@ parent: Building Packages
-    It's much easier to copy an example from the system collection than to write one _de novo_, so you can do
-    
-    ???+ example "On Eagle"
-	     ```
-	     cp /nopt/nrel/apps/modules/default/modulefiles/openmpi/4.0.4/gcc-8.4.0.lua /scratch/$USER/modules/default/openmpi/4.1.0/.
-	     ```
-
-    ???+ example "On Eagle"
-	     ```
-	     cp /nopt/nrel/apps/modules/default/compilers_mpi/openmpi/4.1.5-gcc /scratch/$USER/modules/default/openmpi/4.1.0/.
-	     ```
-    
-    ???+ warning "OpenMpi modulefile on Kestrel"
-         Please note that the OpenMpi modulefile on Kestrel is of TCL type
-         It is not necessary for you to know the language to modify our examples. 
-
-	The Lmod modules system uses the Lua language natively for module code. 
-    Tcl modules will also work under Lmod, but don't offer quite as much flexibility.
-	
-	d. For this example, (a) the OpenMPI version we're building is 4.1.0 instead of 4.0.4 on Eagle or 4.1.5 on Kestrel, and (b) the location is in `/scratch/$USER`, rather than `/nopt/nrel/apps`. 
-    So, edit `/scratch/$USER/modules/default/openmpi/4.1.0/gcc-8.4.0.lua` to make the required changes. 
-    Most of these changes only need to be made at the top of the file; variable definitions take care of the rest.
+Now that the package has been installed to your preferred location, we can set up an environment module.
 
-	e. Now you need to make a one-time change in order to see modules that you put in this collection (`/scratch/$USER/modules/default`). 
-    In your `$HOME/.bash_profile`, add the following line near the top:
+## Create a directory for your module
+If this is your first package, then you probably need to create a place to collect modulefiles. For example, `mkdir -p /scratch/$USER/modules/default`.
 
-	```
-	module use /scratch/$USER/modules/default
-	```
+You can look at the systems module collection(s), _e.g._, `/nopt/nrel/apps/modules/default/modulefiles` on Eagle or `/nopt/nrel/apps/modules/default` on Kestrel, to see how modules are organized from a filesystem perspective. 
+
+In short, each library, application, or framework has its own directory in the `modulefiles` directory, and the modulefile itself sits either in this directory, or one level lower to accomodate additional versioning. 
+
+In this example, there is the MPI version (4.1.0), as well as the compiler type and version (GCC 8.4.0) to keep track of. 
+
+So, we'll make a `/scratch/$USER/modules/default/openmpi/4.1.0` directory, and name the file by the compiler version used to build (gcc-8.4.0). 
+You're free to modify this scheme to suit your own intentions.
+
+## Copy the original modulefile to your new directory
+The modulefiles that we provide are only a starting point. For maximum control, users should copy these files from the locations in /nopt to their own locations for which they have write access.
```

Comment:
This section doesn't quite fit into the example.

Reply:
I am not sure what you mean by this. As in the section header doesn't suit the contents? Or this section is an unnecessary step in creating a modulefile?